src/libsync/propagatedownload.cpp

```diff
@@ -847,7 +847,7 @@ namespace { // Anonymous namespace for the recall feature
 
 void PropagateDownloadFile::transmissionChecksumValidated(const QByteArray &checksumType, const QByteArray &checksum)
 {
-    const auto theContentChecksumType = contentChecksumType();
+    const QByteArray theContentChecksumType = propagator()->account()->capabilities().preferredUploadChecksumType();
```

Comment:
Why did you change this to an explicit type?

Reply:
I love auto but sometimes it helps to see what type you are using :D